ethminer/main.cpp

```diff
@@ -703,8 +703,11 @@ class MinerCLI
 		}
 
 #if API_CORE
-		Api api(this->m_api_port, f);
-        http_server.run(m_http_port, &f, m_show_hwmonitors, m_show_power);
+
+		ApiServer api(m_io_service, abs(m_api_port), (m_api_port < 0) ? true : false, f);
+		api.start();
+
+        if(m_http_port) http_server.run(m_http_port, &f, m_show_hwmonitors, m_show_power);
```

Comment:
If you're going to check for port == 0 here then you don't need to in httpserver.cpp